File: src/main/java/com/exsoloscript/mysteryteams/util/CollectionUtils.java
package com.exsoloscript.mysteryteams.util;

import java.util.ArrayList;
import java.util.List;

public class CollectionUtils {

	public static <T> List<List<T>> split(List<T> toSplit, int howOften) {
		List<List<T>> list = new ArrayList<>(howOften);
		for (int i = 0; i < howOften; i++) {
			list.add(new ArrayList<T>());
		}

		int i = 0;

		for (T t : toSplit) {
			list.get(i).add(t);
			i = (i + 1) % howOften;
		}

		return list;
	}
	
}

File: src/main/kotlin/com/exsoloscript/mysteryteams/util/CollectionUtils.kt
package com.exsoloscript.mysteryteams.util

import java.util.ArrayList

object CollectionUtils {

    fun <T> split(toSplit: List<T>, howOften: Int): List<List<T>> {
        val list = ArrayList<MutableList<T>>(howOften)
        for (i in 0..howOften - 1) {
            list.add(ArrayList<T>())
        }

        var i = 0

        for (t in toSplit) {
            list[i].add(t)
            i = (i + 1) % howOften
        }

        return list
    }

}
